### DISCUSSION TITLE
BIP86 examples?

### DISCUSSION THREAD

**Discussion by miketwenty1** [rust-bitcoin/rust-bitcoin#3282] [Q&A] [answered]
It's unclear if BIP86 is supported similar to BIP32. Are there any good examples of producing HD tapkeys? I'm wanting to quickly create a bunch of m/86 xprv/xpubs.

**Comment by apoelstra** [comment, marked as answer]
Our BIP32 support is sufficent to implement BIP86. But (like with BIPs 49, 84) on principle we do not provide any direct support for it in rust-bitcoin. Output descriptors (as implemented in rust-miniscript) implement a more robust and general solution.

I'm not aware of alternative crates that implement it either, though in most cases with BIPs we decline to support, one does exist somewhere.
